test/unit/model/sharedStimulus/service/StoreServiceTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2014-2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoMediaManager\test\unit\model\sharedStimulus\service;
+
+use oat\generis\test\TestCase;
+use oat\oatbox\filesystem\FileSystem;
+use oat\oatbox\filesystem\FileSystemService;
+use oat\taoMediaManager\model\fileManagement\FlySystemManagement;
+use oat\taoMediaManager\model\sharedStimulus\service\StoreService;
+use Prophecy\Argument;
+
+class StoreServiceTest extends TestCase
+{
+    /**
+     * @var string
+     */
+    private $tempFilePath;
+
+    public function setUp(): void
+    {
+        $this->tempFilePath = '/tmp/' . uniqid();
+        file_put_contents($this->tempFilePath, '');
+    }
+
+    public function tearDown(): void
+    {
+        @unlink($this->tempFilePath);
+    }
+
+    public function testStoreShareStimulus()
+    {
+        $fakeUniqueName = 'uniqueString';
+
+        $fileSystemMock = $this->initFileSystemMock();
+        $fileSystemMock->expects(self::once())
+            ->method('createDir')
+            ->with($fakeUniqueName);
+
+        $fileSystemMock->expects(self::once())
+            ->method('writeStream')
+            ->willReturn(true);
+
+        $stimulusStoreService = $this->getPreparedServiceInstance($fileSystemMock);
+        $stimulusStoreService->expects(self::once())
+            ->method('getUniqueName')
+            ->willReturn($fakeUniqueName);
+
+        $result = $stimulusStoreService->store($this->tempFilePath, 'dummyString', []);
+
+        $this->assertEquals($result, $fakeUniqueName);
+    }
+
+    /**
+     * @return FileSystem|\PHPUnit\Framework\MockObject\MockObject
+     */
+    private function initFileSystemMock()
+    {
+        return $this->getMockBuilder(FileSystem::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['createDir', 'writeStream'])
+            ->getMock();
+    }
+
+    /**
+     * @return StoreService|\PHPUnit\Framework\MockObject\MockObject
+     */
+    private function getPreparedServiceInstance($fileSystemMock)
```

Comment:
In php 7 we have the return types, so we should use. Even if this is a mock, the return type is still `StoreService`.

```suggestion
    private function getPreparedServiceInstance($fileSystemMock): StoreService
```